hydromt/data_catalog.py

```diff
@@ -1594,6 +1595,85 @@ def get_geodataset(
         )
         return obj
 
+    def get_dataset(
+        self,
+        data_like: str | SourceSpecDict | Path | xr.Dataset | xr.DataArray,
+        variables: Optional[List] = None,
+        time_tuple: Optional[Tuple[str] | Tuple[datetime]] = None,
+        single_var_as_array: bool = True,
+        provider: Optional[str] = None,
+        version: Optional[str] = None,
+        **kwargs,
+    ) -> xr.Dataset:
+        """Return a clipped, sliced and unified Dataset.
+
+        To slice the data to the time period of interest, provide the
+        `time_tuple` argument. To return only the dataset variables
+        of interest provide the `variables` argument.
+
+        NOTE: Unless `single_var_as_array` is set to False a single-variable data source
+        will be returned as xarray.DataArray rather than Dataset.
+
+        Arguments
+        ---------
+        data_like: str, Path, xr.Dataset, xr.DataArray
+            Data catalog key, path to geodataset file or geodataset xarray object.
+            The catalog key can be a string or a dictionary with the following keys:
+            {'name', 'provider', 'version'}.
+            If a path to a file is provided it will be added
+            to the catalog with its based on the file basename.
+        time_tuple : tuple of str, datetime, optional
+            Start and end date of period of interest. By default the entire time period
+            of the dataset is returned.
+        single_var_as_array: bool, optional
+            If True, return a DataArray if the dataset consists of a single variable.
+            If False, always return a Dataset. By default True.
+        **kwargs:
+            Additional keyword arguments that are passed to the `DatasetAdapter`
+            function. Only used if `data_like` is a path to a geodataset file.
+
+        Returns
+        -------
+        obj: xarray.Dataset or xarray.DataArray
+            Dataset
+        """
+        if isinstance(data_like, dict):
+            data_like, provider, version = _parse_data_like_dict(
+                data_like, provider, version
+            )
+        if isinstance(data_like, (str, Path)):
+            if isinstance(data_like, str) and data_like in self.sources:
+                name = data_like
+                source = self.get_source(name, provider=provider, version=version)
+            elif os.path.exists(abspath(data_like)):
+                path = str(abspath(data_like))
+                if "provider" not in kwargs:
+                    kwargs.update({"provider": "local"})
```

Comment:
not necessarily true, if I only have one provider but I load it from an `s3` path, it's not a local provider I'd say. If it's indeed a local path this is correct

Reply:
this is still code copied from the get_geodataset method. I see that in the mean time this has changed.

Reply:
I changed it so that is the same as get_geodataset